meshnets/utils/model_training.py

```diff
@@ -90,40 +136,20 @@ def train_model(config):
     # Checkpoint config
     save_top_k = config['save_top_k']
 
-    train_dataset = datasets.load_dataset('inductiva/wind_tunnel',
-                                          version=dataset_version,
-                                          split='train')
-    train_dataset = train_dataset.map(
-        lambda x: data_processing.data_mappers.to_undirected(x, 'edges'))
-    train_dataset = train_dataset.map(
-        data_processing.data_mappers.make_edge_features)
-    train_dataset = train_dataset.map(lambda x: data_processing.data_mappers.
-                                      make_node_features(x, 'wind_vector'))
-
-    train_loader = torch.utils.data.DataLoader(
-        train_dataset,
-        collate_fn=data_processing.torch_utils.dict_to_geometric_data,
-        batch_size=batch_size,
-        num_workers=num_workers_loader,
-        shuffle=True)
+    train_dataset, train_loader = make_dataloader(dataset_version, train_split,
+                                                  'beginning', num_proc,
+                                                  writer_batch_size, batch_size,
+                                                  num_workers_loader)
+    _, val_loader = make_dataloader(dataset_version, val_split, 'end', num_proc,
+                                    writer_batch_size, batch_size,
+                                    num_workers_loader)
 
-    validation_loaders = []
+    validation_loaders = [val_loader]
     for val_dataset_version in val_dataset_versions:
-        val_dataset = datasets.load_dataset('inductiva/wind_tunnel',
-                                            version=val_dataset_version,
-                                            split='train')
-        val_dataset = val_dataset.map(
-            lambda x: data_processing.data_mappers.to_undirected(x, 'edges'))
-        val_dataset = val_dataset.map(
-            data_processing.data_mappers.make_edge_features)
-        val_dataset = val_dataset.map(lambda x: data_processing.data_mappers.
-                                      make_node_features(x, 'wind_vector'))
-
-        validation_loaders.append(
-            torch.utils.data.DataLoader(val_dataset,
-                                        batch_size=batch_size,
-                                        num_workers=num_workers_loader,
-                                        shuffle=False))
+        _, loader = make_dataloader(val_dataset_version, 1, 'beginning',
```

Comment:
Can you comment a bit on this extra validation loaders? I am a bit confused.

Reply:
> Can you comment a bit on this extra validation loaders? I am a bit confused.

This was a feature implemented by Victor. We have the option of giving other datasets for validation. Then, a validation step for each loader will be created and the metrics `val_loss_{specific_dataset_version}` will be logging to mlflow.
That is, this is just a way to compute performance metrics during the validation step on validation datasets other than the one that results from splitting the training data.